core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -200,7 +210,18 @@ private[spark] class AppClient(
 
       case r: RequestExecutors =>
         master match {
-          case Some(m) => context.reply(m.askWithRetry[Boolean](r))
+          case Some(m) =>
+            receiveAndReplyThreadPool.execute(new Runnable {
```

Comment:
this code (the new runnable sending reply) seems to be duplicated a few times, maybe factor it out into a helper function?